reana_job_controller/slurmcern_job_manager.py

```diff
@@ -0,0 +1,168 @@
+        self.slurm_connection = SSHClient(
+            hostname=current_app.config['SLURM_CLUSTER_HOSTNAME'],
+            username=os.environ.get('SLURMCERN_USER'),
+            password=os.environ.get('SLURMCERN_SECRET'),
+            port=current_app.config['SLURM_CLUSTER_PORT'],
+        )
+        self._transfer_inputs()
+        self._dump_job_submission_file()
+        stdin, stdout, stderr = self.slurm_connection.ssh_client.exec_command(
+            'cd {} && sbatch --parsable {}'.format(
+                SlurmJobManagerCERN.SLURM_WORKSAPCE_PATH,
+                self.job_description_file))
+        backend_job_id = stdout.read().decode("utf-8").rstrip()
+        return backend_job_id
+
+    def _dump_job_submission_file(self):
+        """Dump job submission file to the Slurm submit node."""
+        self.job_description_file = 'job_description.sh'
+        job_template = ("#!/bin/bash \n"
+                        "#SBATCH --job-name={job_name} \n"
+                        "#SBATCH --output=reana_job.%N.%j.out \n"
+                        "#SBATCH --error=reana_job.%N.%j.err \n"
+                        "srun {command}".format(
+                            job_name=self.job_name,
+                            command=self._format_cmd()))
+        self.slurm_connection.ssh_client.exec_command(
+            'cd {} && job="{}" && echo "$job" > {}'.format(
+                SlurmJobManagerCERN.SLURM_WORKSAPCE_PATH,
+                job_template,
+                self.job_description_file
+            ))
+
+    def _format_cmd(self):
+        """Return CMD with Slurm submit node absolute paths."""
+        slurm_cmd = []
+        for item in self.cmd:
+            slurm_cmd.append(
+                item.replace(
+                    self.workflow_workspace,
+                    os.path.join(
+                        self.slurm_home_path,
+                        self.workflow_workspace[1:])))
+        # Once RJC is on python 3.8 replace with shlex join()
+        slurm_cmd = "bash -c '{}'".format(' '.join(slurm_cmd[2:]))
+        encoded_slurm_cmd = \
+            base64.b64encode(slurm_cmd.encode('utf-8')).decode('utf-8')
+        return 'echo {}|base64 -d|bash'.format(encoded_slurm_cmd)
+
+    def _get_outputs():
+        """Transfer job outputs to REANA."""
+        os.chdir(SlurmJobManagerCERN.REANA_WORKSPACE_PATH)
+        slurm_connection = SSHClient()
+        sftp = slurm_connection.ssh_client.open_sftp()
+        SlurmJobManagerCERN.download_dir(
+            sftp,
+            SlurmJobManagerCERN.SLURM_WORKSAPCE_PATH,
+            SlurmJobManagerCERN.REANA_WORKSPACE_PATH)
+        sftp.close()
+
+    def download_dir(sftp, remote_dir, local_dir):
+        """Download remote directory content."""
+        os.path.exists(local_dir) or os.makedirs(local_dir)
+        dir_items = sftp.listdir_attr(remote_dir)
+        for item in dir_items:
+            remote_path = remote_dir + '/' + item.filename
```

Comment:
Can't we do `os.path.join` here too? maybe `os.path.join(remote_dir, str(item.filename))` if `item.filename` is some object.